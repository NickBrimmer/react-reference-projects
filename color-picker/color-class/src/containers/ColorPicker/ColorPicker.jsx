import React, { Component } from 'react';
import Controls from '../../components/Controls/Controls';
import Display from '../../components/Display/Display';

export default class ColorPicker extends Component {
  state = {
    color: '#000000',
    backgroundColor: '#FFFFFF',
    text: 'My Very Interesting App'
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {
    const { color, backgroundColor, text } = this.state;
    return (
      <>
        <Controls
          color={color}
          backgroundColor={backgroundColor}
          text={text}
          onChange={this.handleChange}
        />
        <Display
          color={color}
          backgroundColor={backgroundColor}
          text={text}
        />
      </>
    );
  }
}
