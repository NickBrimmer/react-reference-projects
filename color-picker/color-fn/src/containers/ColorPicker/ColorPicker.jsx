import React, { useState } from 'react';
import Display from '../../components/Display/Display';
import Controls from '../../components/Controls/Controls';

export default function ColorPicker() {
  const [color, setColor] = useState('#000000');
  const [backgroundColor, setBackgroundColor] = useState('#FFFFFF');
  const [text, setText] = useState('My Really Great Function Container App');

  const handleChange = ({ target }) => {
    const inputSetterFactory = {
      color: setColor,
      backgroundColor: setBackgroundColor,
      text: setText
    };

    inputSetterFactory[target.name](target.value);

    // switch(target.name) {
    //   case 'color':
    //     setColor(target.value);
    //     break;
    //   case 'backgroundColor':
    //     setBackgroundColor(target.value);
    //     break;
    //   case 'text':
    //     setText(target.value);
    //     break;
    // }
    // if(target.name === 'color') setColor(target.value);
    // if(target.name === 'backgroundColor') setBackgroundColor(target.value);
    // if(target.name === 'text') setText(target.value);
  };

  return (
    <>
      <Controls
        color={color}
        backgroundColor={backgroundColor}
        text={text}
        onChange={handleChange}
      />
      <Display
        color={color}
        backgroundColor={backgroundColor}
        text={text}
      />
    </>
  );
}
