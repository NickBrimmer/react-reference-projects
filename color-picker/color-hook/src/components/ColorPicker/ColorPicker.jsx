import React from 'react';
import Controls from '../Controls/Controls';
import Display from '../Display/Display';
import { useColorPicker } from '../../hooks/color';

export default function ColorPicker() {
  const { color, backgroundColor, text, handleChange } = useColorPicker();

  return (
    <>
      <Controls
        color={color}
        backgroundColor={backgroundColor}
        text={text}
        onChange={handleChange}
      />
      <Display
        color={color}
        backgroundColor={backgroundColor}
        text={text}
      />
    </>
  );
}
