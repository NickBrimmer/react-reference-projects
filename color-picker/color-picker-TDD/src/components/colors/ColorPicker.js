import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class ColorPicker extends Component {
  static propTypes = {
    colors: PropTypes.arrayOf(PropTypes.string),
    backgroundColor: PropTypes.string.isRequired,
    updateColor: PropTypes.func.isRequired
  };

  static defaultProps = {
    colors: ['red', 'blue', 'yellow']
  }

  componentDidMount() {
    console.log('You birthed me!');
  }

  componentWillUnmount() {
    console.log('you killed me!');
  }

  render() {
    const style = {
      border: 0,
      padding: '1rem',
      margin: '0.4rem'
    };

    const buttons = this.props.colors.map(color => (
      <button
        key={color}
        style={{ ...style, backgroundColor: color }}
        onClick={() => this.props.updateColor(color)}>{color}</button>
    ));

    return (
      <>
        {buttons}
        <div style={{ width: '100px', height: '100px', backgroundColor: this.props.backgroundColor }}></div>
      </>
    );
  }
}
