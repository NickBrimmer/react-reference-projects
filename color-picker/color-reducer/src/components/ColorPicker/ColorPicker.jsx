import React from 'react'
import Controls from '../Controls/Controls'
import Display from '../Display/Display'

export default function ColorPicker() {
  return (
    <>
      <Controls />
      <Display />
    </>
  )
}
