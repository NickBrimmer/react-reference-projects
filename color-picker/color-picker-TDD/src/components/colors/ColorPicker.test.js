import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('ColorPicker component', () => {
  it('renders ColorPicker', () => {
    const wrapper = shallow(<ColorPicker />);
    expect(wrapper).toMatchSnapshot();
  });

  it('console logs on color click', () => {
    const wrapper = shallow(<ColorPicker colors={['red', 'green', 'purple']} />);
    wrapper.findWhere(el => el.key() === 'green').simulate('click');
    expect(wrapper.state('backgroundColor')).toBe('green');
  });
});
