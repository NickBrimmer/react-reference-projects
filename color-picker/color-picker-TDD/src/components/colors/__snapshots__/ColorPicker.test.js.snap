// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component renders ColorPicker 1`] = `
<Fragment>
  <button
    key="red"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "red",
        "border": 0,
        "margin": "0.4rem",
        "padding": "1rem",
      }
    }
  >
    red
  </button>
  <button
    key="blue"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "blue",
        "border": 0,
        "margin": "0.4rem",
        "padding": "1rem",
      }
    }
  >
    blue
  </button>
  <button
    key="yellow"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "yellow",
        "border": 0,
        "margin": "0.4rem",
        "padding": "1rem",
      }
    }
  >
    yellow
  </button>
</Fragment>
`;
